File: map_python/src/Source/source_transmitter.py
from dataclasses import dataclass, field

from .skill import Skill
from .source import Source
from .staff import Staff


@dataclass(slots=True)
class Transmitter:
    label: str
    message: str
    object_ref: str
    chief: Staff
    skills: list[Skill] = field(default_factory=list)
    staff_limit: int = 0
    action_limit: int = 0
    value_limit: int = 0


ContainerDestination = str


@dataclass(slots=True)
class SourceTransmitter:
    source: Source
    transmitter: Transmitter
    receiver: ContainerDestination

File: map_ts/src/Source/source-transmitter.ts
// source-transmitter.ts

import type { Source } from "./source.js";
import type { Staff } from "./staff.js";
import type { Skill } from "./skill.js";

export type Transmitter = {
  label: string;
  message: string;

  objectRef: string;

  chief: Staff;

  skills: Skill[];

  staffLimit: number;
  actionLimit: number;
  valueLimit: number;
};

export type ContainerDestination = string;

export type SourceTransmitter = {
  source: Source;
  transmitter: Transmitter;
  receiver: ContainerDestination;
};